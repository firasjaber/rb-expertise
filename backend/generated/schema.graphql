### This file was generated by Nexus Schema
### Do not make changes to this file directly


"""
A date string, such as 2007-12-03, compliant with the `full-date` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
"""
scalar Date

type Employee {
  address: String
  birthDate: Date
  city: String
  createdAt: Date
  email: String
  endDate: Date
  firstName: String
  id: Int!
  isAdmin: Boolean
  lastName: String
  phone: String
  region: String
  startDate: Date
}

type Query {
  employee(employeeId: Int!): Employee
  employees(searchQuery: String): [Employee]
  me: String
}

type User {
  id: Int
  username: String
}
