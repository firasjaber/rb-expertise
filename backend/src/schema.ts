import { makeSchema } from 'nexus';
import { nexusPrisma } from 'nexus-plugin-prisma';
import { join } from 'path';
import * as types from './graphql';

export const schema = makeSchema({
  types,
  outputs: {
    typegen: join(__dirname, '..', 'nexus-typegen.ts'),
    schema: join(__dirname, '..', 'schema.graphql'),
  },
  contextType: {
    module: join(__dirname, './context.ts'),
    export: 'Context',
  },
  plugins: [nexusPrisma()],
});
