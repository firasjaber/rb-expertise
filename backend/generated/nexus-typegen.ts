/**
 * This file was generated by Nexus Schema
 * Do not make changes to this file directly
 */


import type { Context } from "./../src/context"
import type { core } from "nexus"
declare global {
  interface NexusGenCustomInputMethods<TypeName extends string> {
    /**
     * A date string, such as 2007-12-03, compliant with the `full-date` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
     */
    date<FieldName extends string>(fieldName: FieldName, opts?: core.CommonInputFieldConfig<TypeName, FieldName>): void // "Date";
  }
}
declare global {
  interface NexusGenCustomOutputMethods<TypeName extends string> {
    /**
     * A date string, such as 2007-12-03, compliant with the `full-date` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
     */
    date<FieldName extends string>(fieldName: FieldName, ...opts: core.ScalarOutSpread<TypeName, FieldName>): void // "Date";
  }
}


declare global {
  interface NexusGen extends NexusGenTypes {}
}

export interface NexusGenInputs {
  AppointmentCreateInput: { // input type
    assuranceId: number; // Int!
    date: NexusGenScalars['Date']; // Date!
    employeeId: number; // Int!
    location: string; // String!
    notes?: string | null; // String
    title: string; // String!
  }
  AppointmentResolveInput: { // input type
    id: number; // Int!
    resolved: boolean; // Boolean!
  }
  EmployeeCreateInput: { // input type
    address: string; // String!
    birthDate: NexusGenScalars['Date']; // Date!
    city: string; // String!
    email: string; // String!
    endDate?: NexusGenScalars['Date'] | null; // Date
    firstName: string; // String!
    lastName: string; // String!
    phone: string; // String!
    pictureUrl?: string | null; // String
    region: string; // String!
    startDate: NexusGenScalars['Date']; // Date!
  }
  EmployeeUpdateInput: { // input type
    address: string; // String!
    birthDate: NexusGenScalars['Date']; // Date!
    city: string; // String!
    email: string; // String!
    endDate?: NexusGenScalars['Date'] | null; // Date
    firstName: string; // String!
    id: number; // Int!
    lastName: string; // String!
    phone: string; // String!
    pictureUrl?: string | null; // String
    region: string; // String!
    startDate: NexusGenScalars['Date']; // Date!
  }
}

export interface NexusGenEnums {
}

export interface NexusGenScalars {
  String: string
  Int: number
  Float: number
  Boolean: boolean
  ID: string
  Date: any
}

export interface NexusGenObjects {
  Appointment: { // root type
    assuranceId: number; // Int!
    createdAt: NexusGenScalars['Date']; // Date!
    date: NexusGenScalars['Date']; // Date!
    employeeId: number; // Int!
    id: number; // Int!
    location?: string | null; // String
    notes?: string | null; // String
    resolved?: boolean | null; // Boolean
    resolvedAt?: NexusGenScalars['Date'] | null; // Date
    title: string; // String!
  }
  Assurance: { // root type
    id: number; // Int!
    name?: string | null; // String
    pictureUrl?: string | null; // String
    slug?: string | null; // String
  }
  Employee: { // root type
    address?: string | null; // String
    birthDate?: NexusGenScalars['Date'] | null; // Date
    city?: string | null; // String
    createdAt?: NexusGenScalars['Date'] | null; // Date
    email?: string | null; // String
    endDate?: NexusGenScalars['Date'] | null; // Date
    firstName?: string | null; // String
    id: number; // Int!
    isAdmin?: boolean | null; // Boolean
    lastName?: string | null; // String
    phone?: string | null; // String
    pictureUrl?: string | null; // String
    region?: string | null; // String
    startDate?: NexusGenScalars['Date'] | null; // Date
  }
  Mutation: {};
  Query: {};
  User: { // root type
    id?: number | null; // Int
    username?: string | null; // String
  }
}

export interface NexusGenInterfaces {
}

export interface NexusGenUnions {
}

export type NexusGenRootTypes = NexusGenObjects

export type NexusGenAllTypes = NexusGenRootTypes & NexusGenScalars

export interface NexusGenFieldTypes {
  Appointment: { // field return type
    assurance: NexusGenRootTypes['Assurance'] | null; // Assurance
    assuranceId: number; // Int!
    createdAt: NexusGenScalars['Date']; // Date!
    date: NexusGenScalars['Date']; // Date!
    employee: NexusGenRootTypes['Employee'] | null; // Employee
    employeeId: number; // Int!
    id: number; // Int!
    location: string | null; // String
    notes: string | null; // String
    resolved: boolean | null; // Boolean
    resolvedAt: NexusGenScalars['Date'] | null; // Date
    title: string; // String!
  }
  Assurance: { // field return type
    appointments: Array<NexusGenRootTypes['Appointment'] | null> | null; // [Appointment]
    id: number; // Int!
    name: string | null; // String
    pictureUrl: string | null; // String
    slug: string | null; // String
  }
  Employee: { // field return type
    address: string | null; // String
    appointments: Array<NexusGenRootTypes['Appointment'] | null> | null; // [Appointment]
    birthDate: NexusGenScalars['Date'] | null; // Date
    city: string | null; // String
    createdAt: NexusGenScalars['Date'] | null; // Date
    email: string | null; // String
    endDate: NexusGenScalars['Date'] | null; // Date
    firstName: string | null; // String
    id: number; // Int!
    isAdmin: boolean | null; // Boolean
    lastName: string | null; // String
    phone: string | null; // String
    pictureUrl: string | null; // String
    region: string | null; // String
    startDate: NexusGenScalars['Date'] | null; // Date
  }
  Mutation: { // field return type
    createAppointment: NexusGenRootTypes['Appointment'] | null; // Appointment
    createOneEmployee: NexusGenRootTypes['Employee'] | null; // Employee
    deleteOneEmployee: boolean | null; // Boolean
    resolveAppointment: NexusGenRootTypes['Appointment'] | null; // Appointment
    updateOneEmployee: NexusGenRootTypes['Employee'] | null; // Employee
  }
  Query: { // field return type
    appointment: NexusGenRootTypes['Appointment'] | null; // Appointment
    appointments: Array<NexusGenRootTypes['Appointment'] | null> | null; // [Appointment]
    assurances: Array<NexusGenRootTypes['Assurance'] | null> | null; // [Assurance]
    employee: NexusGenRootTypes['Employee'] | null; // Employee
    employees: Array<NexusGenRootTypes['Employee'] | null> | null; // [Employee]
    me: string | null; // String
  }
  User: { // field return type
    id: number | null; // Int
    username: string | null; // String
  }
}

export interface NexusGenFieldTypeNames {
  Appointment: { // field return type name
    assurance: 'Assurance'
    assuranceId: 'Int'
    createdAt: 'Date'
    date: 'Date'
    employee: 'Employee'
    employeeId: 'Int'
    id: 'Int'
    location: 'String'
    notes: 'String'
    resolved: 'Boolean'
    resolvedAt: 'Date'
    title: 'String'
  }
  Assurance: { // field return type name
    appointments: 'Appointment'
    id: 'Int'
    name: 'String'
    pictureUrl: 'String'
    slug: 'String'
  }
  Employee: { // field return type name
    address: 'String'
    appointments: 'Appointment'
    birthDate: 'Date'
    city: 'String'
    createdAt: 'Date'
    email: 'String'
    endDate: 'Date'
    firstName: 'String'
    id: 'Int'
    isAdmin: 'Boolean'
    lastName: 'String'
    phone: 'String'
    pictureUrl: 'String'
    region: 'String'
    startDate: 'Date'
  }
  Mutation: { // field return type name
    createAppointment: 'Appointment'
    createOneEmployee: 'Employee'
    deleteOneEmployee: 'Boolean'
    resolveAppointment: 'Appointment'
    updateOneEmployee: 'Employee'
  }
  Query: { // field return type name
    appointment: 'Appointment'
    appointments: 'Appointment'
    assurances: 'Assurance'
    employee: 'Employee'
    employees: 'Employee'
    me: 'String'
  }
  User: { // field return type name
    id: 'Int'
    username: 'String'
  }
}

export interface NexusGenArgTypes {
  Mutation: {
    createAppointment: { // args
      data: NexusGenInputs['AppointmentCreateInput']; // AppointmentCreateInput!
    }
    createOneEmployee: { // args
      data: NexusGenInputs['EmployeeCreateInput']; // EmployeeCreateInput!
    }
    deleteOneEmployee: { // args
      employeeId: number; // Int!
    }
    resolveAppointment: { // args
      data: NexusGenInputs['AppointmentResolveInput']; // AppointmentResolveInput!
    }
    updateOneEmployee: { // args
      data: NexusGenInputs['EmployeeUpdateInput']; // EmployeeUpdateInput!
    }
  }
  Query: {
    appointment: { // args
      appointmentId: number; // Int!
    }
    appointments: { // args
      searchQuery?: string | null; // String
    }
    employee: { // args
      employeeId: number; // Int!
    }
    employees: { // args
      searchQuery?: string | null; // String
    }
  }
}

export interface NexusGenAbstractTypeMembers {
}

export interface NexusGenTypeInterfaces {
}

export type NexusGenObjectNames = keyof NexusGenObjects;

export type NexusGenInputNames = keyof NexusGenInputs;

export type NexusGenEnumNames = never;

export type NexusGenInterfaceNames = never;

export type NexusGenScalarNames = keyof NexusGenScalars;

export type NexusGenUnionNames = never;

export type NexusGenObjectsUsingAbstractStrategyIsTypeOf = never;

export type NexusGenAbstractsUsingStrategyResolveType = never;

export type NexusGenFeaturesConfig = {
  abstractTypeStrategies: {
    isTypeOf: false
    resolveType: true
    __typename: false
  }
}

export interface NexusGenTypes {
  context: Context;
  inputTypes: NexusGenInputs;
  rootTypes: NexusGenRootTypes;
  inputTypeShapes: NexusGenInputs & NexusGenEnums & NexusGenScalars;
  argTypes: NexusGenArgTypes;
  fieldTypes: NexusGenFieldTypes;
  fieldTypeNames: NexusGenFieldTypeNames;
  allTypes: NexusGenAllTypes;
  typeInterfaces: NexusGenTypeInterfaces;
  objectNames: NexusGenObjectNames;
  inputNames: NexusGenInputNames;
  enumNames: NexusGenEnumNames;
  interfaceNames: NexusGenInterfaceNames;
  scalarNames: NexusGenScalarNames;
  unionNames: NexusGenUnionNames;
  allInputTypes: NexusGenTypes['inputNames'] | NexusGenTypes['enumNames'] | NexusGenTypes['scalarNames'];
  allOutputTypes: NexusGenTypes['objectNames'] | NexusGenTypes['enumNames'] | NexusGenTypes['unionNames'] | NexusGenTypes['interfaceNames'] | NexusGenTypes['scalarNames'];
  allNamedTypes: NexusGenTypes['allInputTypes'] | NexusGenTypes['allOutputTypes']
  abstractTypes: NexusGenTypes['interfaceNames'] | NexusGenTypes['unionNames'];
  abstractTypeMembers: NexusGenAbstractTypeMembers;
  objectsUsingAbstractStrategyIsTypeOf: NexusGenObjectsUsingAbstractStrategyIsTypeOf;
  abstractsUsingStrategyResolveType: NexusGenAbstractsUsingStrategyResolveType;
  features: NexusGenFeaturesConfig;
}


declare global {
  interface NexusGenPluginTypeConfig<TypeName extends string> {
  }
  interface NexusGenPluginInputTypeConfig<TypeName extends string> {
  }
  interface NexusGenPluginFieldConfig<TypeName extends string, FieldName extends string> {
  }
  interface NexusGenPluginInputFieldConfig<TypeName extends string, FieldName extends string> {
  }
  interface NexusGenPluginSchemaConfig {
  }
  interface NexusGenPluginArgConfig {
  }
}