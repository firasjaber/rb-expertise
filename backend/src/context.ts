import { prisma } from './db';

import { PrismaClient } from '@prisma/client';

export interface Context {
  prisma: PrismaClient;
}
export const context = {
  prisma,
};
