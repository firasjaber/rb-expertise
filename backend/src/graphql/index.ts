export { GQLDate } from './modules/shared';
export * from './modules/user';
export * from './modules/employee';
export * from './modules/appointment';
