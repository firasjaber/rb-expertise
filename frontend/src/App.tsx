import './App.css';
import { Button, ChakraProvider } from '@chakra-ui/react';

function App() {
  return (
    <ChakraProvider>
      <h3 className='text-sm'>hello word</h3>
      <Button>supsup</Button>
    </ChakraProvider>
  );
}

export default App;
