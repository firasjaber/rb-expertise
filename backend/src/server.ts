import { ApolloServer } from 'apollo-server';
import { ApolloServerPluginLandingPageGraphQLPlayground } from 'apollo-server-core';
import { context } from './context';
import { schema } from './schema';

export const server = new ApolloServer({
  schema,
  context,
  plugins: [ApolloServerPluginLandingPageGraphQLPlayground()],
});
