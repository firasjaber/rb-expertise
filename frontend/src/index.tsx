import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';

const client = new ApolloClient({
  uri: 'http://localhost:4000/graphql',
  cache: new InMemoryCache(),
});

ReactDOM.render(
  <React.StrictMode>
    <ApolloProvider client={client}>
      <App />
    </ApolloProvider>
  </React.StrictMode>,
  document.getElementById('root')
);
