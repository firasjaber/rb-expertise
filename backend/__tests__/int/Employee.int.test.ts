import { ApolloServer, gql } from 'apollo-server-express';
import { NexusGenInputs } from '../../generated/nexus-typegen';
import { context } from '../../src/context';
import { apolloServerConfig } from '../../src/server';
import {
  CREATE_EMPLOYEE,
  CREATE_EMPLOYEE_TO_UPDATE,
  DELETE_EMPLOYEE,
  GET_ALL_EMPLOYEES,
  UPDATE_EMPLOYEE,
} from '../config/graphql';

describe('Employee tests', () => {
  let server: ApolloServer;
  server = new ApolloServer(apolloServerConfig);

  afterAll(async () => {
    await context.prisma.employee.deleteMany();
    await context.prisma.$disconnect();
  });
  it('get all employees query should return empty when the db is emplty', async () => {
    const res = await server.executeOperation({
      query: GET_ALL_EMPLOYEES,
    });
    expect(res.data).toBeDefined();
    expect(res.data?.employees).toStrictEqual([]);
  });
  it('create employee mutation: invalid input', async () => {
    //missing required inputs (eg : lastName,birthDate,startDate,)
    const mockEmployee = {
      address: 'Groove street, Earth, 1000',
      city: 'City',
      region: 'Earth',
      email: 'jhon@gmail.com',
      firstName: 'Jhon',
      phone: '10200300',
    };
    const res = await server.executeOperation({
      query: CREATE_EMPLOYEE,
      variables: { createOneEmployeeData: mockEmployee },
    });
    expect(res.errors).toBeDefined();
    expect(res.errors?.length).toBe(3);
    expect(res.errors?.[0].extensions?.code).toBe('BAD_USER_INPUT');
    expect(res.errors?.[1].extensions?.code).toBe('BAD_USER_INPUT');
    expect(res.errors?.[2].extensions?.code).toBe('BAD_USER_INPUT');
  });
  it('create employee mutation: valid input', async () => {
    const mockEmployee: NexusGenInputs['EmployeeCreateInput'] = {
      firstName: 'Jhon',
      lastName: 'Doe',
      email: 'jhon@gmail.com',
      birthDate: '2001-02-19',
      phone: '10200300',
      address: 'Groove street, Earth, 1000',
      city: 'City',
      region: 'Earth',
      startDate: '2021-12-01',
    };
    const res = await server.executeOperation({
      query: CREATE_EMPLOYEE,
      variables: { createOneEmployeeData: mockEmployee },
    });
    expect(res.data).toBeDefined();
    expect(res.data).toMatchSnapshot();
  });
  it('create employee mutation: already exists email', async () => {
    const mockEmployee: NexusGenInputs['EmployeeCreateInput'] = {
      firstName: 'Jhon',
      lastName: 'Doe',
      email: 'jhon@gmail.com',
      birthDate: '2001-02-19',
      phone: '10200300',
      address: 'Groove street, Earth, 1000',
      city: 'City',
      region: 'Earth',
      startDate: '2021-12-01',
    };
    const res = await server.executeOperation({
      query: CREATE_EMPLOYEE,
      variables: { createOneEmployeeData: mockEmployee },
    });
    expect(res.data).toBeDefined();
    expect(res.errors).toMatchSnapshot();
  });
  it('edit employee mutation: invalid input', async () => {
    //create valid employee then test against it
    const mockEmployee: NexusGenInputs['EmployeeCreateInput'] = {
      firstName: 'Jhon',
      lastName: 'Doe',
      email: 'jhon1@gmail.com',
      birthDate: '2001-02-19',
      phone: '10200300',
      address: 'Groove street, Earth, 1000',
      city: 'City',
      region: 'Earth',
      startDate: '2021-12-01',
    };
    const createRes = await server.executeOperation({
      query: CREATE_EMPLOYEE_TO_UPDATE,
      variables: { createOneEmployeeData: mockEmployee },
    });
    // removing firstName and phone from the input so it should return an input error
    const { firstName, phone, ...rest } = createRes.data?.createOneEmployee;
    const udpatedMockEmployee = { ...rest };
    const res = await server.executeOperation({
      query: UPDATE_EMPLOYEE,
      variables: { updateOneEmployeeData: udpatedMockEmployee },
    });
    expect(res.errors).toBeDefined();
    expect(res.errors?.length).toBe(2);
    expect(res.errors?.[0].extensions?.code).toBe('BAD_USER_INPUT');
    expect(res.errors?.[1].extensions?.code).toBe('BAD_USER_INPUT');
    //expect(res.errors).toMatchSnapshot();
  });
  it('edit employee mutation: not existing employee', async () => {
    //create valid employee then test against it
    const mockEmployee: NexusGenInputs['EmployeeCreateInput'] = {
      firstName: 'Jhon',
      lastName: 'Doe',
      email: 'jhon3@gmail.com',
      birthDate: '2001-02-19',
      phone: '10200300',
      address: 'Groove street, Earth, 1000',
      city: 'City',
      region: 'Earth',
      startDate: '2021-12-01',
    };
    const createRes = await server.executeOperation({
      query: CREATE_EMPLOYEE_TO_UPDATE,
      variables: { createOneEmployeeData: mockEmployee },
    });
    const { id, ...rest } = createRes.data?.createOneEmployee;
    const udpatedMockEmployee = { ...rest, id: 1000000, firstName: 'Jhonny', phone: '20400500' };
    const res = await server.executeOperation({
      query: UPDATE_EMPLOYEE,
      variables: { updateOneEmployeeData: udpatedMockEmployee },
    });
    expect(res.errors).toBeDefined();
    expect(res.errors).toMatchSnapshot();
  });
  it('edit employee mutation: valid input', async () => {
    //create valid employee then test against it
    const mockEmployee: NexusGenInputs['EmployeeCreateInput'] = {
      firstName: 'Jhon',
      lastName: 'Doe',
      email: 'jhon2@gmail.com',
      birthDate: '2001-02-19',
      phone: '10200300',
      address: 'Groove street, Earth, 1000',
      city: 'City',
      region: 'Earth',
      startDate: '2021-12-01',
    };
    const createRes = await server.executeOperation({
      query: CREATE_EMPLOYEE_TO_UPDATE,
      variables: { createOneEmployeeData: mockEmployee },
    });
    const { firstName, phone, ...rest } = createRes.data?.createOneEmployee;
    const udpatedMockEmployee = { ...rest, firstName: 'Jhonny', phone: '20400500' };
    const res = await server.executeOperation({
      query: UPDATE_EMPLOYEE,
      variables: { updateOneEmployeeData: udpatedMockEmployee },
    });
    expect(res.data).toBeDefined();
    expect(res.data).toMatchSnapshot();
  });
  it('delete employee mutation: invalid input', async () => {
    //this mutation needs only id so this shouldnt work
    const res = await server.executeOperation({
      query: DELETE_EMPLOYEE,
      variables: { firstName: 'jhon' },
    });
    expect(res.errors).toBeDefined();
    expect(res.errors).toMatchSnapshot();
  });
  it('delete employee mutation: not existing employee input', async () => {
    const res = await server.executeOperation({
      query: DELETE_EMPLOYEE,
      variables: { deleteOneEmployeeId: 1000000 },
    });
    expect(res.errors).toBeDefined();
    expect(res.errors).toMatchSnapshot();
  });
  it('delete employee mutation: valid input', async () => {
    //create valid employee then test against it
    const mockEmployee: NexusGenInputs['EmployeeCreateInput'] = {
      firstName: 'Jhon',
      lastName: 'Doe',
      email: 'jhon4@gmail.com',
      birthDate: '2001-02-19',
      phone: '10200300',
      address: 'Groove street, Earth, 1000',
      city: 'City',
      region: 'Earth',
      startDate: '2021-12-01',
    };
    const createRes = await server.executeOperation({
      query: CREATE_EMPLOYEE_TO_UPDATE,
      variables: { createOneEmployeeData: mockEmployee },
    });
    const { id } = createRes.data?.createOneEmployee;
    const res = await server.executeOperation({
      query: DELETE_EMPLOYEE,
      variables: { deleteOneEmployeeId: id },
    });
    expect(res.data).toBeDefined();
    expect(res.data?.deleteOneEmployee).toBe(true);
  });

  it('get all employees query return array list', async () => {
    const res = await server.executeOperation({
      query: GET_ALL_EMPLOYEES,
    });
    expect(res.data?.employees?.length).toBeGreaterThan(0);
  });
});
