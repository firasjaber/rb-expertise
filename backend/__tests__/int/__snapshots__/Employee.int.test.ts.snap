// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee tests create employee mutation: already exists email 1`] = `
Array [
  [GraphQLError: Email already exists for another employee],
]
`;

exports[`Employee tests create employee mutation: valid input 1`] = `
Object {
  "createOneEmployee": Object {
    "address": "Groove street, Earth, 1000",
    "birthDate": "2001-02-19",
    "city": "City",
    "createdAt": "2021-10-12",
    "email": "jhon@gmail.com",
    "endDate": null,
    "firstName": "Jhon",
    "isAdmin": false,
    "lastName": "Doe",
    "phone": "10200300",
    "pictureUrl": "https://imgproxy.generated.photos/qZKLWxkKQ-IwCzW5AYoDKbjn9GUT6aGuJ-b918HaEJE/rs:fit:256:256/czM6Ly9pY29uczgu/Z3Bob3Rvcy1wcm9k/LnBob3Rvcy90cmFu/c3BhcmVudF92My92/M18wNDk4NDQ4LnBu/Zw.png",
    "region": "Earth",
    "startDate": "2021-12-01",
  },
}
`;

exports[`Employee tests delete employee mutation: invalid input 1`] = `
Array [
  [UserInputError: Variable "$deleteOneEmployeeId" of required type "Int!" was not provided.],
]
`;

exports[`Employee tests delete employee mutation: not existing employee input 1`] = `
Array [
  [GraphQLError: Employee not found to delete],
]
`;

exports[`Employee tests edit employee mutation: not existing employee 1`] = `
Array [
  [GraphQLError: Employee not found to update],
]
`;

exports[`Employee tests edit employee mutation: valid input 1`] = `
Object {
  "updateOneEmployee": Object {
    "address": "Groove street, Earth, 1000",
    "birthDate": "2001-02-19",
    "city": "City",
    "createdAt": "2021-10-12",
    "email": "jhon2@gmail.com",
    "endDate": null,
    "firstName": "Jhonny",
    "isAdmin": false,
    "lastName": "Doe",
    "phone": "20400500",
    "pictureUrl": "https://imgproxy.generated.photos/qZKLWxkKQ-IwCzW5AYoDKbjn9GUT6aGuJ-b918HaEJE/rs:fit:256:256/czM6Ly9pY29uczgu/Z3Bob3Rvcy1wcm9k/LnBob3Rvcy90cmFu/c3BhcmVudF92My92/M18wNDk4NDQ4LnBu/Zw.png",
    "region": "Earth",
    "startDate": "2021-12-01",
  },
}
`;
